### PR TITLE
chore: refactor `Region` to `GraphRegion`

### PR DESCRIPTION
**Pull request by Wdestroier**
Hi! I solved the name clash and changed the Region class name to GraphRegion as suggested.

**Note: On merging this, `Lost`'s references to `PolyWorld::Region` will break and need to be adjusted to `GraphRegion`.**

### CODE REVIEW

**Review by jdrueckert — CHANGES_REQUESTED** [review]
Hi @Wdestroier,
thanks for the PR.
I believe you missed a few spots though. For instance, `LevelSpawnSystem` still imports `org.terasology.polyworld.graph.Region` which, expectedly, it cannot find anymore now.
Please adjust these spots as well :slightly_smiling_face:

**Comment by Wdestroier** [comment]
Hi! @jdrueckert I found the LevelSpawnSystem class in the org.terasology.lost package, am I missing any dependency?

**Comment by jdrueckert** [comment]
@Wdestroier Yes you're absolutely right, sorry about that, my brain must be a bit gooey already :roll_eyes: 
I just wanted to test your PR and saw that it doesn't compile. But that's because `Lost` depends on `PolyWorld` and uses the `Region`. So in order to not break the `Lost` gameplay we'd need to adjust it to the change in here.
Are you up for adding another PR and then we can merge both?

**Review by jdrueckert — APPROVED** [review]

**Comment by jdrueckert** [comment]
So we had another potential contributor ask for taking https://github.com/Terasology/PolyWorld/issues/24 (ping @gaurav9822). I mentioned that they maybe can take on the `Lost` side of this.

File: src/main/java/org/terasology/polyworld/IslandWorldGenerator.java [hunk 1]
```diff
@@ -102,11 +102,11 @@ public Vector3f getSpawnPosition(EntityRef entity) {
         GraphFacet graphs = worldRegion.getFacet(GraphFacet.class);
         WhittakerBiomeModelFacet model = worldRegion.getFacet(WhittakerBiomeModelFacet.class);
         Vector2f pos2d = new Vector2f(pos.getX(), pos.getZ());
-        CirclePickerClosest<org.terasology.polyworld.graph.Region> picker = new CirclePickerClosest<>(pos2d);
+        CirclePickerClosest<org.terasology.polyworld.graph.GraphRegion> picker = new CirclePickerClosest<>(pos2d);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/polyworld/IslandWorldGenerator.java:105]: This should now be possible without the fully qualified name (as there is no name clash anymore)
```suggestion
        CirclePickerClosest<GraphRegion> picker = new CirclePickerClosest<>(pos2d);
```
This change would likely need an import, though.

File: src/main/java/org/terasology/polyworld/IslandWorldGenerator.java [hunk 2]
```diff
@@ -102,11 +102,11 @@ public Vector3f getSpawnPosition(EntityRef entity) {
...
 
         for (Graph g : graphs.getAllGraphs()) {
             BiomeModel biomeModel = model.get(g);
-            for (org.terasology.polyworld.graph.Region r : g.getRegions()) {
+            for (org.terasology.polyworld.graph.GraphRegion r : g.getRegions()) {
```
Comment (skaldarnar) [inline src/main/java/org/terasology/polyworld/IslandWorldGenerator.java:109]:
```suggestion
            for (GraphRegion r : g.getRegions()) {
```

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**